src/tests.cpp

```diff
@@ -195,38 +222,1038 @@ class HostConnectorTests : public QObject
+        assert_return(checkOnlyConnectionBothWays(blockPortOut1(1, 2), blockPortIn2(0, 4)), false);
+        assert_return(checkOnlyConnectionBothWays(blockPortOut2(1, 2), blockPairPortIn2(0, 4)), false);
+        assert_return(testNoPassthrough(), false);
+
+        // move stereo block to row 0 between the sidechaining blocks
+        assert_return(connector.reorderBlock(1, 2, 5), false);
+        assert_return(connector.swapBlockRow(1, 5, 0, 5), false);
+        assert_return(connector.reorderBlock(0, 5, 3), false);
+        // row 0 connections
+        assert_return(checkOnlyConnection(blockPortIn1(0, 1), JACK_CAPTURE_PORT_1), false);
+        assert_return(checkOnly2Connections(blockPortOut1(0, 1), blockPortIn1(0, 3), blockPortIn2(0, 3)), false);
+        assert_return(checkOnlyConnection(blockPortIn1(0, 3), blockPortOut1(0, 1)), false);
+        assert_return(checkOnlyConnection(blockPortIn2(0, 3), blockPortOut1(0, 1)), false);
+        assert_return(checkOnlyConnectionBothWays(blockPortOut1(0, 3), blockPortIn1(0, 5)), false);
+        assert_return(checkOnlyConnectionBothWays(blockPortOut2(0, 3), blockPairPortIn1(0, 5)), false);
+        assert_return(checkOnlyConnection(blockPortOut1(0, 5), JACK_PLAYBACK_PORT_1), false);
+        assert_return(checkOnlyConnection(blockPairPortOut1(0, 5), JACK_PLAYBACK_PORT_2), false);
+        // row 1 connections
+        assert_return(checkOnly2Connections(blockPortOut2(0, 1), blockPortIn2(0, 5), blockPairPortIn2(0, 5)), false);
+        assert_return(checkOnlyConnection(blockPortIn2(0, 5), blockPortOut2(0, 1)), false);
+        assert_return(checkOnlyConnection(blockPairPortIn2(0, 5), blockPortOut2(0, 1)), false);
+        assert_return(testNoPassthrough(), false);
+
+        // remove all blocks
+        assert_return(connector.replaceBlock(0, 5, nullptr), false);
+        assert_return(connector.replaceBlock(0, 3, nullptr), false);
+        assert_return(connector.replaceBlock(0, 1, nullptr), false);
+
+        return true;
+    }
+
+    bool testSideChain() 
+    {
+        // NOTE: even though this test set is extensive, it doesn't include every possible scenario
+        // focus is on issues that have appeared earlier in development
+
+        // NOTE: many of the connection checks in between actions are the same within one test
+        // and between test functions but the steps to get there are different
+
+        assert_return(testSideChainBuiltInOrder(), false);
+        assert_return(testPassthrough(), false);
+
+        assert_return(testSideChainAddStereoInBetween(), false);
+        assert_return(testPassthrough(), false);
+
+        assert_return(testSideChainMoveStereoOnFirstRow(), false);
+        assert_return(testPassthrough(), false);
+        
+        assert_return(testSideChainSwapBlockRows(), false);
+        assert_return(testPassthrough(), false);
+
+        return true;
+    }
+
+
+    // HELPERS
+
+    bool checkNoConnections(std::string port_to_check)
```

Comment:
these functions are all converting to string list where not really needed... can be improved, but I can do that afterwards